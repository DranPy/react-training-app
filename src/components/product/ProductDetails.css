.breadcrumbs ul {
  display: flex;
}

.breadcrumbs ul li .svg-inline--fa {
  margin-right: 10px;
  margin-left: 10px;
}


.product-details h1 {
  margin-bottom: 0;
}

.product-details .product-details__author {
  margin-bottom: 20px;
  display: block;
}

.product-details .product-details__price {
  font-size: 20px;
}