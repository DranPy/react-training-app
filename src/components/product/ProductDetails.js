import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Loader from '../Loader';

class ProductDetails extends Component {
  static propTypes = {
    isLoading: PropTypes.bool,
    product: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      category: PropTypes.string,
      price: PropTypes.string,
      imgUrl: PropTypes.string,
    }),
    fetchProduct: PropTypes.func.isRequired,
  };

  componentDidMount() {
    const id = this.props.match.params.id;
    this.props.fetchProduct(id);
  }

  render() {
    const { isLoading, product } = this.props;
    const { name, category, price, imgUrl } = product;

    return (
      <div className="product-details">
        <div className="container">
          {!isLoading && product ? (
            <div>
              <h1>{name}</h1>
              <div>
                <img src={imgUrl} className="img-fluid" alt={name} />
              </div>
              <div>{price}</div>
            </div>
          ) : (
            <Loader />
          )}
        </div>
      </div>
    );
  }
}

export default ProductDetails;
