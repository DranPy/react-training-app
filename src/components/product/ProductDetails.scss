.breadcrumbs {
  ul { 
    display: flex;  

    li { 
      .svg-inline--fa  {
        margin-right: 10px;
        margin-left: 10px;
      }
    }
  }
}

.product-details {
  h1 {
    margin-bottom: 0;
  }

  &__author {
    margin-bottom: 20px;
    display: block;
  }

  &__price {
    font-size: 20px;
  }
}