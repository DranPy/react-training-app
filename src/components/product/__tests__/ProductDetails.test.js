import { shallow } from 'enzyme';
import React from 'react';

import ProductDetails from '../ProductDetails';

describe('<ProductDetails />', () => {
  const product = {
    id: '1',
    name: 'test name 1',
    category: 'category 1',
    price: '1$',
    imgUrl: 'src/1.img',
  };

  const defaultProps = {
    match: {
      params: {
        id: product.id,
      },
    },
    fetchProduct: () => {},
    isLoading: false,
    product,
  };

  it(`should trigger fetchProduct function with argument '${product.id}'`, () => {
    const fetchProductHandler = jest.fn();
    shallow(<ProductDetails {...defaultProps} fetchProduct={fetchProductHandler} />);
    expect(fetchProductHandler).toHaveBeenCalledTimes(1);
    expect(fetchProductHandler).toHaveBeenCalledWith(product.id);
  });

  it('should render product details', () => {
    const wrapper = shallow(<ProductDetails {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render product details', () => {
    const wrapper = shallow(<ProductDetails {...defaultProps} isLoading={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
