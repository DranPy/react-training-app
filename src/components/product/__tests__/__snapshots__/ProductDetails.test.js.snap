// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetails /> should not render product details 1`] = `
<div
  className="product-details"
>
  <div
    className="container"
  >
    <Loader />
  </div>
</div>
`;

exports[`<ProductDetails /> should render product details 1`] = `
<div
  className="product-details"
>
  <div
    className="container"
  >
    <div>
      <h1>
        test name 1
      </h1>
      <div>
        <img
          alt="test name 1"
          className="img-fluid"
          src="src/1.img"
        />
      </div>
      <div>
        1$
      </div>
    </div>
  </div>
</div>
`;
